src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -322,8 +322,12 @@ private void ClearModifiedNestedBlocks(List<AssociativeNode> astNodes)
 
                     // Remove from the global codeblocks
                     core.CodeBlockList.RemoveAll(x => x.guid == bnode.guid);// && x.AstID == bnode.OriginalAstID);
+
                     // Remove from the runtime codeblocks
-                    core.CompleteCodeBlockList.RemoveAll(x => x.guid == bnode.guid);// && x.AstID == bnode.OriginalAstID);
+                    foreach (var key in core.CompleteCodeBlockDict.Where(x => x.Value.guid == bnode.guid).Select(x => x.Key).ToArray())
+                    {
+                        core.CompleteCodeBlockDict.Remove(key);
```

Comment:
How about doing this:
```
core.CompleteCodeBlockDict = core.CompleteCodeBlockDict
  .Where(kv => !(kv.Value.guid == bnode.guid))
  .ToDictionary(kv => kv.Key, kv => kv.Value);
```

Reply:
Hmm..your suggestions does seem to have less operations 
I'll give it a try. THanks

Reply:
Hmm...actually  completeCodeBlockDict is a SortedDictionary.
Your code would have to do something like
`
core.CompleteCodeBlockDict = new SortedDictionary<int, CodeBlock>(core.CompleteCodeBlockDict.Where(kv => !(kv.Value.guid == bnode.guid)).ToDictionary(kv => kv.Key, kv => kv.Value));
`
It might not be worth it (creates a standard dictionary and allocates another SortedDict) also I find it a bit less obvious then the original.

To make it even more obvious/ easy to understand maybe this would work better:
`
var keysToRemove = core.CompleteCodeBlockDict.Where(x => x.Value.guid == bnode.guid).Select(x => x.Key).ToArray();
keysToRemove.ForEach(key => core.CompleteCodeBlockDict.Remove(key));
`

I don't think performance should be an issue here...